src/lem/slot.rs

```diff
@@ -233,16 +233,27 @@ impl Block {
     }
 }
 
+#[derive(Clone, Debug)]
+pub enum Val {
+    Pointer(Ptr),
+    Num(usize),
+    Boolean(bool),
+}
```

Comment:
Maybe `SlotDataVal` would be more explicit, but what I think is really missing is documentation. For example, the `Num` variant needs some explanation for why it holds a `usize`